### ISSUE TITLE
Comments on WebView injected script will break the scipt

### ISSUE THREAD

**Issue by mishuagopian** [facebook/react-native#9749]
### Issue Description

When injecting a script to a `WebView` component via `injectedJavaScript` prop, if the script contains any comment then it won't work on android.
### Steps to Reproduce / Code Snippets

Create a `WebView` in which we inject a script with a simple comment in it.
#### Code to reproduce

``` es6
import React from 'react';
import { AppRegistry, StyleSheet, Text, View, WebView } from 'react-native';

function SampleApp () {
  const html = `
    <html>
      <body>
        <h1 class="title"> Hello! </h1>
      </body>
    </html>
  `;
  const script = `
    var text = "Bye!"; // This comment will break the script on android
    document.querySelector(".title").textContent = text;
  `;
  return (
    <View style={styles.container}>
      <Text> WebView script comment issue </Text>
      <WebView
        source={{html: html}}
        injectedJavaScript={script}
        javaScriptEnabled
      />  
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'stretch',
    backgroundColor: '#F5FCFF',
    paddingTop: 50
  }
});

AppRegistry.registerComponent('SampleApp', () => SampleApp);
```

[Working example](https://rnplay.org/apps/TWZWYA)
#### Expected Results

Text on html `h1` element ("Hello!") should be replaced with "Bye!" text.
#### Actual Results

Text on `h1` html element changes on iOS but not on Android. Removing the comment on the js script will fix this issue.
### Additional Information
- React Native version: 0.31.0
- Platform(s) (iOS, Android, or both?): both
- Operating System (macOS, Linux, or Windows?): macOS

**Comment by eremzeit** [comment]
I can confirm.

**Comment by lacker** [comment]
Weird. I wonder if some compression is happening which removes the endline and thus turns the subsequent line into comment as well. Are you interested in sending a PR to fix this?

**Comment by mishuagopian** [comment]
@lacker sure!
Here you are: https://github.com/facebook/react-native/pull/12321.

**Comment by rgoldiez** [comment]
I faced the same issue on Android only (works fine on iOS).  It wasn't until I ran across this issue that I was about to remove comments in the injected javascript to get my WebView to work on Android.

**Comment by hellogerard** [comment]
This is happening for me on React Native 0.43 on Android (iOS is working). I found that switching from double-slash-type comments to C-style comments works.

```
// these comments won't work
...
/* but, these comments will */
```

**Comment by hramos** [comment]
Hi there! This issue is being closed because it has been inactive for a while. Maybe the issue has been fixed in a recent release, or perhaps it is not affecting a lot of people. Either way, we're automatically closing issues after a period of inactivity. Please do not take it personally!

If you think this issue should definitely remain open, please let us know. The following information is helpful when it comes to determining if the issue should be re-opened:

- Does the issue still reproduce on the latest release candidate? Post a comment with the version you tested.
- If so, is there any information missing from the bug report? Post a comment with all the information required by the [issue template](https://github.com/facebook/react-native/blob/master/.github/ISSUE_TEMPLATE.md).
- Is there a pull request that addresses this issue? Post a comment with the PR number so we can follow up.

If you would like to work on a patch to fix the issue, *contributions are very welcome*! Read through the [contribution guide](http://facebook.github.io/react-native/docs/contributing.html), and feel free to hop into [#react-native](https://discordapp.com/invite/0ZcbPKXt5bZjGY5n) if you need help planning your contribution.

**Comment by grabbou** [comment]
I've been constantly experiencing this issue and it looks to me like a valid bug to be fixed. It fails silently when you include comments in your code and makes it really hard to debug. In my opinion, should be reopened.

**Comment by hramos** [comment]
Reopening. Please take a look at the proposed fix at #12321.

**Comment by fungilation** [comment]
I'm still seeing this issue.
